#! /bin/sh /usr/share/dpatch/dpatch-run
## types_list.dpatch by Sebastian Harl <sh@tokkee.org>
##
## DP: configfile, types_list:
## DP: Improved error messages when handling "TypesDB".

@DPATCH@

diff a/src/types_list.c b/src/types_list.c
--- a/src/types_list.c
+++ b/src/types_list.c
@@ -172,7 +172,9 @@ int read_types_list (const char *file)
   if (fh == NULL)
   {
     char errbuf[1024];
-    ERROR ("open (%s) failed: %s", 
+    fprintf (stderr, "Failed to open types database `%s': %s.\n",
+	file, sstrerror (errno, errbuf, sizeof (errbuf)));
+    ERROR ("Failed to open types database `%s': %s",
 	file, sstrerror (errno, errbuf, sizeof (errbuf)));
     return (-1);
   }
